eclair-node/src/main/scala/fr/acinq/eclair/api/handlers/Message.scala

```diff
@@ -41,8 +43,16 @@ trait Message {
   }
 
   val sendOnionMessage: Route = postRequest("sendonionmessage") { implicit t =>
-    formFields("route".as[List[PublicKey]](pubkeyListUnmarshaller), "content".as[ByteVector](binaryDataUnmarshaller), "pathId".as[ByteVector](binaryDataUnmarshaller)?) { (route, userCustomContent, pathId) =>
-      complete(eclairApi.sendOnionMessage(route, userCustomContent, pathId))
+    formFields(
+      "destination".as[Either[OnionMessages.Recipient, Sphinx.RouteBlinding.BlindedRoute]](destinationUnmarshaller),
```

Comment:
From an API's point of view, I think it's confusing to use the same field for two very distinct cases (and having to match on its length to decide which case we're using). What about using two distinct arguments (e.g. `recipientNode` and `recipientBlindedRoute`) and requiring that exactly one is provided (like we do in some places where we want either an invoice to be provided or an explicit payment hash)?

For the caller apps, I believe it's cleaner as well as it maps to separate use-cases (e.g. messaging a given nodeId vs fetching an invoice through offers).